script/hipify-inplace.sh

```diff
@@ -51,23 +61,36 @@ echo "Logs written to ${HIPIFY_LOG}"
 
 # Additional fixes for project-specific things and things hipify misses or gets
 # wrong.
-for src in ${srcs[@]}; do
-    sed -i 's@#include <hipblas.h>@#include <hipblas/hipblas.h>@' $src
-    sed -i 's@#include <hipsparse.h>@#include <hipsparse/hipsparse.h>@' $src
-    sed -i 's@#include <cuda_fp8.h>@#include <hip/hip_fp8.h>@' $src
-    sed -i 's@#include <cuda_bf16.h>@#include <hip/hip_bf16.h>@' $src
-    sed -i 's@\bDGL_USE_CUDA\b@DGL_USE_ROCM@g' $src
-    sed -i 's@\bCUB_VERSION\b@HIPCUB_VERSION@g' $src
-    sed -i 's@\bCUDART_ZERO_BF16\b@HIPRT_ZERO_BF16@g' $src
-    sed -i 's@\bCUDART_INF_BF16\b@HIPRT_INF_BF16@g' $src
-    sed -i 's@\bthrust::cuda::par@thrust::hip::par@g' $src
-    sed -i 's@\b__nv_fp8_e4m3\b@__hip_fp8_e4m3@g' $src
-    sed -i 's@\b__nv_fp8_e5m2\b@__hip_fp8_e5m2@g' $src
-    sed -i 's@\bCUBLAS_GEMM_DEFAULT_TENSOR_OP\b@HIPBLAS_GEMM_DEFAULT@g' $src
-    sed -i 's@\bcurand4\b@hiprand4@g' $src
-    # hipify uses the old one
-    sed -i 's@\bhip_bfloat16\b@__hip_bfloat16@g' $src
-    sed -i 's@\b__trap();@abort();@' $src
+for src in ${all_srcs[@]}; do
+    sed -i -e 's@#include <hipblas.h>@#include <hipblas/hipblas.h>@' \
+        -e 's@#include <hipsparse.h>@#include <hipsparse/hipsparse.h>@' \
+        -e 's@#include <cuda_fp8.h>@#include <hip/hip_fp8.h>@' \
+        -e 's@#include <cuda_bf16.h>@#include <hip/hip_bf16.h>@' \
+        -e 's@#include <cub/cub.cuh>@#include <hipcub/hipcub.hpp>@' \
+        -e 's@#include "hip/extension/@#include "./cuda/extension/@' \
+        -e 's@#include "hip/cooperative_minibatching_utils.h"@#include "./cuda/cooperative_minibatching_utils.h"@' \
+        -e 's@#include "hip/max_uva_threads.h"@#include "./cuda/max_uva_threads.h"@' \
+        -e 's@\.\./hip/@@' \
+        -e 's@\bDeviceFor::Bulk\b@Bulk@g' \
+        -e 's@\bcub::@hipcub::@g' \
+        -e 's@\bcuda::stream_ref@cuco::cuda_stream_ref@' \
+        -e 's@\bcuda::::min@min@' \
+        -e 's@\bcuda::proclaim_return_type@proclaim_return_type@' \
+        -e 's@\bDGL_USE_CUDA\b@DGL_USE_ROCM@g' \
+        -e 's@\b__CUDA_ARCH__\b@__HIP_DEVICE_COMPILE__@g' \
```

Comment:
I thought I'd commented on this previously, but I don't think we want to do this blanket replace here. `__CUDA_ARCH__` is a number that is used for feature detection, not just testing for a device compile. When I wanted to unconditionally enable for HIP, I modified the source code to add a check for `__HIP_DEVICE_COMIPLE__` as well (e.g. https://github.com/nod-ai/dgl/blob/hip-ready/src/array/cuda/atomic.cuh#L18) and then if you can figure out what actual feature is needed, the right way is `__HIP_ARCH_<FEATURE>__`

See https://rocm.docs.amd.com/projects/HIP/en/latest/how-to/hip_porting_guide.html#distinguishing-compilers-and-platforms

Reply:
Whoops, good point. I've updated the code and removed this line.